api/internal/localizer/localizer.go

```diff
@@ -76,8 +85,62 @@ func (lc *Localizer) Localize() error {
 }
 
 // TODO(annasong): implement
-// processKust returns a copy of the kustomization at kt with paths localized.
-func (lc *Localizer) processKust(kt *target.KustTarget) *types.Kustomization {
-	kust := kt.Kustomization()
-	return &kust
+// localizeFields localizes paths on the kustomization's built-in fields, excluding helm.
+func (lc *Localizer) localizeFields(_ *types.Kustomization) error {
+	return nil
+}
+
+// localizePlugins localizes built-in plugins on kust that can contain file paths. The built-in plugins
+// can be inline or a file, but not the result of a kustomization.
+func (lc *Localizer) localizePlugins(kust *types.Kustomization) error {
+	for field, plugins := range map[string][]string{
+		"generator":   kust.Generators,
+		"transformer": kust.Transformers,
+		"validator":   kust.Validators,
+	} {
+		for i, entry := range plugins {
+			var isPath bool
+			rm, err := lc.rFactory.NewResMapFromBytes([]byte(entry))
+			if err != nil {
+				rm, err = lc.rFactory.FromFile(lc.ldr, entry)
+				if err != nil {
+					return errors.WrapPrefixf(err, "%q is neither a valid inline nor file path %s", entry, field)
+				}
+				isPath = true
+			}
+			localizedPlugin, err := lc.localizePluginEntry(rm)
+			if err != nil {
+				return errors.WrapPrefixf(err, "unable to localize %s entry %q", field, entry)
+			}
+			var newEntry string
+			if isPath {
+				// TODO(annasong): write localizedPlugin to dst
+				newEntry = entry
+			} else {
+				newEntry = string(localizedPlugin)
+			}
+			plugins[i] = newEntry
+		}
+	}
+	return nil
+}
+
+// localizePluginEntry localizes pluginEntry, the resources in a plugin entry, and returns
+// the localized pluginEntry in bytes if they are built-ins that can specify file paths.
+// Otherwise, localizePluginEntry returns an error.
+func (lc *Localizer) localizePluginEntry(pluginEntry resmap.ResMap) ([]byte, error) {
+	localizedPlugins := make([][]byte, pluginEntry.Size())
+	for i, plugin := range pluginEntry.Resources() {
+		if !isBuiltinPlugin(plugin) {
+			return nil, errors.Errorf("plugin is not built-in")
+		}
+		content, err := plugin.AsYAML()
+		if err != nil {
+			return nil, errors.WrapPrefixf(err, "unable to serialize plugin in entry")
+		}
+		// TODO(annasong): localize plugins
+		localizedPlugins[i] = content
+	}
+	localizedEntry := bytes.Join(localizedPlugins, []byte("---\n"))
+	return bytes.TrimSuffix(localizedEntry, []byte("\n")), nil
 }
```

Comment:
Suggest leveraging `ResMap` or `Resource` to formalize the code. Maybe @natasha41575 can give some guidance and  evaluate whether this could be simpler. 
Here's some pseudocode. 
```
func (lc *Localizer) localizePluginEntry(pluginEntry resmap.ResMap) ([]byte, error) {
  newPlugins []*resource.Resource
  for _, plugin := range pluginEntry.Resources() {
     newPlugin := plugin.DeepCopy()
     filter := LocalizeBuiltinPlugin{}
     if err := newPlugin.ApplyFilter(LocalizeBuiltinPlugin{}); err != nil {
         return ...
     }
     newPlugins = append(newPlugins, newPlugin)
   }
   newPluginEntry := lc.rFactory.FromResourceSlice(newPlugins)
   return newPluginEntry.AsYaml()
}

type LocalizeBuiltinPlugin struct {}
func (f *LocalizeBuiltinPlugin) Filter(rnodes []*yaml.RNode) ([]*yaml.RNode, error) {
    if len(rnodes) > 0 {
           return nil, err
    }
    
    if rnodes[0].GetApiVersion != konfig.BuiltinPluginApiVersion {
          // not builtin plugin config
          return rnodes, nil
     } 
    fnSpec, err := runtimeutil.GetFunctionSpec(rnodes[0])
    // localize the path from fnSpec
}
```

Reply:
I need to look into this. Thanks for the suggestion, @yuwenma! This is very eye-opening.

Reply:
I got started on this approach. The rest of this pseudocode will be implemented in different PRs. Thanks again, @yuwenma!

Reply:
I really like how the code looks.